File: apps/api/src/docflow/domain/documents.py
from enum import StrEnum


class DocumentStatus(StrEnum):
    UPLOADED = "uploaded"
    PREPROCESSING = "preprocessing"
    CLASSIFYING = "classifying"
    EXTRACTING = "extracting"
    VALIDATING = "validating"
    NEEDS_REVIEW = "needs_review"
    APPROVED = "approved"
    REJECTED = "rejected"
    FAILED = "failed"
    DUPLICATE_FILE = "duplicate_file"


class ProcessingRunStatus(StrEnum):
    RUNNING = "running"
    SUCCEEDED = "succeeded"
    AWAITING_OCR = "awaiting_ocr"
    FAILED = "failed"


class RevisionStatus(StrEnum):
    NEEDS_REVIEW = "needs_review"
    APPROVED = "approved"


TERMINAL_STATUSES = {
    DocumentStatus.REJECTED,
    DocumentStatus.DUPLICATE_FILE,
}

File: apps/web/src/lib/documents.ts
export const statusLabels: Record<string, string> = {
  uploaded: "Загружен",
  preprocessing: "Предобработка",
  classifying: "Классификация",
  extracting: "Извлечение",
  validating: "Проверка правил",
  needs_review: "Ручная проверка",
  approved: "Утверждён",
  rejected: "Отклонён",
  failed: "Ошибка",
  duplicate_file: "Точный дубль",
};

export function statusLabel(status: string) {
  return statusLabels[status] ?? status.replaceAll("_", " ");
}

export function formatBytes(bytes: number) {
  if (bytes < 1024 * 1024) return `${Math.max(1, Math.round(bytes / 1024))} КБ`;
  return `${(bytes / 1024 / 1024).toFixed(1)} МБ`;
}

export function formatDate(value: string, includeTime = true) {
  return new Intl.DateTimeFormat("ru-RU", {
    day: "2-digit",
    month: "short",
    year: "numeric",
    ...(includeTime ? { hour: "2-digit", minute: "2-digit" } : {}),
  }).format(new Date(value));
}
